xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java

```diff
@@ -190,6 +191,10 @@ private void handleClusterDiscovered() {
         lbProvider = lbRegistry.getProvider("ring_hash");
         lbConfig = new RingHashConfig(root.result.minRingSize(), root.result.maxRingSize());
       }
+      if (root.result.lbPolicy() == LbPolicy.LEAST_REQUEST) {
+        lbProvider = lbRegistry.getProvider("least_request_experimental");
+        lbConfig = new LeastRequestConfig(root.result.choiceCount());
```

Comment:
We should restrict the choiceCount to 10 _somewhere_ (e.g., here, ClientXdsClient, or even LeastRequestConfig). Yes, this is from xDS so is more trustworthy, but Java is the only language that makes these objects directly. The other languages create a service config snippet and have them parsed. So we should keep the behavior of LeastRequestConfig consistent for both flows, because other languages will do the same.

Reply:
Right now restricts choiceCount to 10 in `parseLoadBalancingPolicyConfig` but I think it also makes sense to do it in LeastRequestConfig as well to cover all possible cases. Will add something tomorrow.